woocommerce/woocommerce
# [Settings] Add custom field type to data

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/52319



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add/Enable Gutenberg and turn on `settings` feature flag
2. WooCommerce > Settings
3. In the console, look for email's custom type in the array of its settings

```js
window.wcSettings.admin.settingsData.email.sections.default.settings
```
4. There should be a custom type which reflects its custom html:

https://github.com/woocommerce/woocommerce/blob/9239decd1da9a22a23237af0e906fb09a1cd5b58/plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php#L611-L617



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Add Settings custom type handling to reflect custom html

</details>
